response.go

```diff
@@ -11,21 +12,54 @@ type Response interface {
 	Header() Header
 	Decode() ([]interface{}, error)
 	DecodeTyped(res interface{}) error
-
-	Pos() []byte
-	MetaData() []ColumnMetaData
-	SQLInfo() SQLInfo
 }
 
-type ConnResponse struct {
+type BaseResponse struct {
```

Comment:
Is there a reason to make it public?
```suggestion
type baseResponse struct {
```

Reply:
Before we created mocks, we needed to use some Responses in the tests.